### DISCUSSION TITLE
Getting the effective color from styles in code.

### DISCUSSION THREAD

**Discussion by JPfahl** [dotnet/maui#25865] [Q&A]
I am setting the luminosity of the ImageSource in an ImageButton to give the user visual feedback that the button was pressed using the Pressed, Released and Unfocus events.  Even though I restore the color to its original value when the press is complete, when the theme changes the color is not set to the new color from the new theme.  I am guessing that under the covers the styles setup a binding that is lost when I set the color in code.  Is there a way to restore the binding?  Is there a way to force the style to apply the color when the theme changes?  is there a way to get the effective color from the styles so I can set it myself in code?  I use both styles applied using TargetType and styles set using StyleClass.

**Comment by stephenquan** [comment]
[Style apps using XAML](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/styles/xaml?view=net-maui-8.0) has a section on Style Inheritance

> Styles can inherit from other styles to reduce duplication and enable reuse. This is achieved by setting the Style.BasedOn property to an existing [Style](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.style). In XAML, this can be achieved by setting the BasedOn property to a [StaticResource](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.xaml.staticresourceextension) markup extension that references a previously created [Style](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.style).

You can declare [Visual States](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/visual-states?view=net-maui-8.0) for Normal, Pressed and Focused states and use these visual states to switch between your various styles.

**Reply by JPfahl** [reply]
I know all that.  I use styles extensively but I cannot use a visual state in this case because I am not setting the property to a set value but one calculated from the effective color applied to each image source.  So I ask again: Is there a way to restore the binding for a style? Is there a way to force the style to apply the color when the theme changes? is there a way to get the effective color from the styles so I can set it myself in code?

**Reply by stephenquan** [reply]
I don't fully understand your use case. Some code snippets will go along way to help clarify what you want.

Meanwhile, consider putting a control with that style in your ResourceDictionary. Then, you can just query the colors directly from those controls, for instance:

```xaml
<ContentPge>
    <ContentPage.Resources>
        <ResourceDictionary>
            <Label x:Key="HeadlineLabel" Style="{StaticResource Headline}" />
        </ResourceDictionary>
    </ContentPage.Resources>
    <!-- ... -->
        <Label TextColor="{Binding TextColor, Source={StaticResource HeadlineLabel}}" />
    <!-- ... -->
</ContentPage>
```

**Reply by JPfahl** [reply]
Again the color I'm setting is not a static value it is calculated at runtime per instance.   So, is there a way to get the effective color from the styles so I can set it myself in code?  Once a property value is set in code it is orphaned from the styles. So when the theme changes the property is not updated by the style.

**Comment by JPfahl** [comment]
I have found a different way to solve this problem.
